tests/tests.js

```diff
@@ -364,59 +364,55 @@ QUnit.test('processText: nested escaped characters', function(assert) {
   );
 });
 
-QUnit.test('processPseudoElements: element with id', function(assert) {
+QUnit.test('processPseudoElements', function(assert) {
   var serializer = new HTMLSerializer();
   var fixture = document.getElementById('qunit-fixture');
   var element = document.createElement('div');
-  element.setAttribute('id', 'myID');
   var style = document.createElement('style');
   style.appendChild(document.createTextNode('div::before{content:"test";}'));
   fixture.appendChild(style);
   fixture.appendChild(element);
-  serializer.processPseudoElements(element);
+  serializer.processPseudoElements(element, 'myId');
   var styleText = window.getComputedStyle(element, ':before').cssText;
-  assert.equal(serializer.html.length, 0);
   assert.equal(serializer.pseudoElementCSS.length, 1);
-  assert.equal(serializer.pseudoElementCSS[0], `#myID::before{${styleText}} `);
+  assert.equal(serializer.pseudoElementCSS[0], `#myId::before{${styleText}} `);
 });
 
-QUnit.test('processPseudoElements: element without id', function(assert) {
+QUnit.test('processTree: element without id', function(assert) {
   var serializer = new HTMLSerializer();
   var fixture = document.getElementById('qunit-fixture');
   var element = document.createElement('div');
-  var style = document.createElement('style');
-  style.appendChild(document.createTextNode('div::after{content:"test";}'));
-  fixture.appendChild(style);
   fixture.appendChild(element);
-  serializer.processPseudoElements(element);
-  var styleText = window.getComputedStyle(element, ':after').cssText;
-  assert.equal(serializer.html[0], 'id="snap-it0" ');
-  assert.equal(serializer.pseudoElementCSS.length, 1);
-  assert.equal(
-    serializer.pseudoElementCSS[0],
-    `#snap-it0::after{${styleText}} `
-  );
+  serializer.processTree(element);
+  assert.equal(serializer.html[0], '<div ');
+  assert.equal(serializer.html[2], 'id="snap-it0" ');
+  assert.equal(serializer.html[3], '>');
+});
+
+QUnit.test('processTree: element with id', function(assert) {
+  var serializer = new HTMLSerializer();
+  var fixture = document.getElementById('qunit-fixture');
+  var element = document.createElement('div');
+  element.setAttribute('id', 'myId');
+  fixture.appendChild(element);
+  serializer.processTree(element);
+  assert.equal(serializer.html[0], '<div ');
```

Comment:
I don't think I'd be able to test for this with a unit test, because HTMLSerializer still produces a document with large styles.  It's only in popup.js that the styles are minimized.  Should I try to write a simple test that goes through the whole process of using HTMLSerializer and then popup?